src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java

```diff
@@ -14,37 +16,17 @@
 
 public class ConfigurableExtensionFilter extends AbstractDescribableImpl<ConfigurableExtensionFilter> {
 
-    @Extension
-    public static final ExtensionFilter EXTENSION_FILTER = new ExtensionFilter() {
-
-        @Override
-        public <T> boolean allows(Class<T> tClass, ExtensionComponent<T> tExtensionComponent) {
-            // Don't exclude myself
-            if (tExtensionComponent.getInstance() == this) return true;
-
-            return DESCRIPTOR.allows(tClass.getName())
-                    && DESCRIPTOR.allows(
-                            tExtensionComponent.getInstance().getClass().getName());
-        }
-    };
-
-    @Extension
-    public static final DescriptorVisibilityFilter DESCRIPTOR_FILTER = new DescriptorVisibilityFilter() {
-
-        @Override
-        public boolean filter(@CheckForNull Object context, @Nonnull Descriptor descriptor) {
-            Class contextClass = context == null ? null : context.getClass();
-            return DESCRIPTOR.allows(contextClass, descriptor.getClass().getName());
-        }
-    };
-
-    @Extension
-    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
+    public static /*almost final*/ DescriptorImpl DESCRIPTOR;
 
     @Symbol("configurableExtensionFilter")
+    @Extension
     public static final class DescriptorImpl extends Descriptor<ConfigurableExtensionFilter> {
 
```

Comment:
```
static {
            DESCRIPTOR = new DescriptorImpl();
        }
```
initialize DESCRIPTOR field before it is accessed

Reply:
I think this need some test. My first feeling is that Jenkins will load all extension even if they are ignored.

We need to test disabling an extension (for example freestyle job) and see if they persist after a reload and reboot.

Perhaps also a test with RestartableJenkinsRule can validate this.

If you are motivated you can take the code from this PR and update the code based and your comment and provide test if possible